bokehjs/src/lib/models/tools/edit/freehand_draw_tool.ts

```diff
@@ -0,0 +1,110 @@
+import {Keys} from "core/dom"
+import {UIEvent, GestureEvent, TapEvent, KeyEvent} from "core/ui_events"
+import * as p from "core/properties"
+import {isArray} from "core/util/types"
+import {EditTool, EditToolView} from "./edit_tool"
+
+export class FreehandDrawToolView extends EditToolView {
+  model: FreehandDrawTool
+
+  _draw(ev: UIEvent, mode: string, emit: boolean = false): void {
+    if (!this.model.active) { return; }
+
+    const renderer = this.model.renderers[0];
```

Comment:
If the tool can only operate on one renderer (which seems sensible, offhand) then the API should reflect that

Reply:
I wanted to maintain consistency with other drawing tools, which generally allow drawing to only the first renderer but do allow deleting and moving glyphs on multiple renderers. In this particular case you can only delete items on multiple renderers so I'd definitely be happy to change it.

Reply:
After investigating this I think we should either stay consistent and keep this as a list or change all the editing tools, except for the PolyEditTool, to accept only a single renderer. That would be an API change and I'd have to do a fair bit of refactoring to make it work, so if you think we should go ahead with that I'd probably prefer merging this as is and doing the refactoring of all the tools at once in a new PR.

Reply:
Just to be clear, it's not a difficult change, I would just want to do it all at once since it's tied up with changes to the base class.

Reply:
Changing everything seems disruptive at this point, so let's just make sure the documentation is excessively clear about expected behaviours

Reply:
Okay fine with me. I'm expecting to make two more drawing tool related PRs after this one:

1. Add support for vertex snapping to the poly draw and edit tools
2. Final cleanup and tests for all edit tools

I'd propose that we merge this PR and I add that those clarifications in 2.